probe-rs-cli-util/src/logging.rs

```diff
@@ -115,27 +192,179 @@ pub fn clear_progress_bar() {
+        "Your error was reported successfully. If you don't mind, please open an issue on Github and include the UUID: ",
+        uuid
+    ));
+}
+
+/// Captures an std::error::Error with sentry and sends all previously captured logs.
+pub fn capture_error<E>(metadata: &Metadata, error: &E)
+where
+    E: Error + ?Sized,
+{
+    let _guard = sentry::init(sentry_config(metadata.release.clone()));
+    set_metadata(metadata);
+    send_logs();
+    let uuid = sentry::capture_error(error);
+    print_uuid(uuid);
+}
+
+/// Captures an anyhow error with sentry and sends all previously captured logs.
+pub fn capture_anyhow(metadata: &Metadata, error: &anyhow::Error) {
+    let _guard = sentry::init(sentry_config(metadata.release.clone()));
+    set_metadata(metadata);
+    send_logs();
+    let uuid = sentry::integrations::anyhow::capture_anyhow(error);
+    print_uuid(uuid);
+}
+
+/// Captures a panic with sentry and sends all previously captured logs.
+pub fn capture_panic(metadata: &Metadata, info: &PanicInfo<'_>) {
+    let _guard = sentry::init(sentry_config(metadata.release.clone()));
+    set_metadata(metadata);
+    send_logs();
+    let uuid = sentry::capture_event(PanicIntegration::new().event_from_panic_info(info));
+    print_uuid(uuid);
+}
+
+/// Ask for a line of text.
+fn text() -> std::io::Result<String> {
+    // Read up to the first newline or EOF.
+
+    let mut out = String::new();
+    std::io::stdin().read_line(&mut out)?;
+
+    // Only capture up to the first newline.
+    if let Some(mut newline) = out.find('\n') {
+        if newline > 0 && out.as_bytes()[newline - 1] == b'\r' {
+            newline -= 1;
+        }
+        out.truncate(newline);
+    }
+
+    Ok(out)
+}
+
+/// Displays the text to ask if the crash should be reported.
+pub fn ask_to_log_crash() -> bool {
+    if let Ok(var) = std::env::var("PROBE_RS_SENTRY") {
+        var == "true"
+    } else {
+        println(format!(
+            "        {} {}",
+            "Hint".blue().bold(),
+            "Unfortunately probe-rs encountered an unhandled problem. To help the devs, you can automatically log the error to sentry.technokrat.ch."
+        ));
+        println(format!(
+            "             {}",
+            "Your data will be transmitted completely anonymous and cannot be associated with you directly."
+        ));
+        println(format!(
+            "             {}",
+            "To Hide this message in the future, please set $PROBE_RS_SENTRY to 'true' or 'false'."
+        ));
+        print!("             {}", "Do you wish to transmit the data? y/N: ");
+        std::io::stdout().flush().ok();
+        let result = if let Ok(s) = text() {
+            let s = s.to_lowercase();
+            if s.is_empty() {
+                false
+            } else if "yes".starts_with(&s) {
+                true
+            } else if "no".starts_with(&s) {
+                false
+            } else {
+                false
+            }
```

Comment:
```suggestion
            } else {
                false
            }
```

I agree with clippy here, the check for "no" doesn't seem necessary.

Reply:
I do not agree with clippy. In this case maybe, yes. But not in the general case. Complicated conditions are way worse than more branches :)